pkg/dynamicaccess/accesslogic.go

```diff
@@ -0,0 +1,157 @@
+// Copyright 2024 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package dynamicaccess
+
+import (
+	"context"
+	"crypto/ecdsa"
+	"fmt"
+
+	"github.com/ethersphere/bee/v2/pkg/encryption"
+	"github.com/ethersphere/bee/v2/pkg/kvs"
+	"github.com/ethersphere/bee/v2/pkg/swarm"
+	"golang.org/x/crypto/sha3"
+)
+
+//nolint:gochecknoglobals
+var (
+	hashFunc      = sha3.NewLegacyKeccak256
+	oneByteArray  = []byte{1}
+	zeroByteArray = []byte{0}
+)
+
+// Decryptor is a read-only interface for the ACT.
+type Decryptor interface {
+	// DecryptRef will return a decrypted reference, for given encrypted reference and grantee
+	DecryptRef(ctx context.Context, storage kvs.KeyValueStore, encryptedRef swarm.Address, publisher *ecdsa.PublicKey) (swarm.Address, error)
+	Session
+}
+
+// Control interface for the ACT (does write operations).
+type Control interface {
+	Decryptor
+	// AddGrantee adds a new grantee to the ACT
+	AddGrantee(ctx context.Context, storage kvs.KeyValueStore, publisherPubKey, granteePubKey *ecdsa.PublicKey) error
+	// EncryptRef encrypts a Swarm reference for a given grantee
+	EncryptRef(ctx context.Context, storage kvs.KeyValueStore, grantee *ecdsa.PublicKey, ref swarm.Address) (swarm.Address, error)
+}
+
+type ActLogic struct {
+	Session
+}
+
+var _ Control = (*ActLogic)(nil)
+
+// EncryptRef encrypts a SWARM reference for a publisher.
+func (al ActLogic) EncryptRef(ctx context.Context, storage kvs.KeyValueStore, publisherPubKey *ecdsa.PublicKey, ref swarm.Address) (swarm.Address, error) {
+	accessKey, err := al.getAccessKey(ctx, storage, publisherPubKey)
+	if err != nil {
+		return swarm.ZeroAddress, err
+	}
+	refCipher := encryption.New(accessKey, 0, uint32(0), hashFunc)
```

Comment:
No need to type the literal 0. `uint(0)` -> `0`. This happens on 10 other places, change in all places.

Reply:
done